opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py

```diff
@@ -479,7 +488,10 @@ def _receive_metrics(
             return
         token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
         try:
-            self._exporter.export(metrics_data, timeout_millis=timeout_millis)
+            with self._export_lock:
```

Comment:
I've implemented this in PeriodicExportingMetricReader because it's the only thing that uses MetricExporter in the spec. I think this satisfies [the requirement](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exportbatch:~:text=Export%20will%20never%20be%20called%20concurrently%20for%20the%20same%20exporter%20instance.%20Export%20can%20be%20called%20again%20only%20after%20the%20current%20call%20returns.).

The other option is to lock in the MetricReader when calling `self._receive_metrics()` https://github.com/open-telemetry/opentelemetry-python/blob/506300083e4b58947f8de64a9d42be5757fdccdd/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py#L312

but this would be stronger guarantee than what is in the spec. IMO it's not necessary. Any thoughts?

Reply:
>The other option is to lock in the MetricReader when calling self._receive_metrics() 

+1. While we only have the `PeriodicExportingMetricReader` that is exporting now, I think the requirement needs to be enforced for any implementation of `MetricReader` that exports?

Reply:
> The other option is to lock in the MetricReader when calling self._receive_metrics()

Wouldn't this lock any future `MetricReader`s that do not use an exporter to be not be able to be called concurrently? (maybe this is fine?)

> I think the requirement needs to be enforced for any implementation of MetricReader that exports?

I think "THAT EXPORTS" is the key. Not sure how we can verify that from the `MetricReader` class though.

Reply:
> Wouldn't this lock any future `MetricReader`s that do not use an exporter to be not be able to be called concurrently? (maybe this is fine?)

Ya that's what I was getting at. I don't think it would have much impact, so I could go either way

Reply:
>Wouldn't this lock any future MetricReaders that do not use an exporter to be not be able to be called concurrently? (maybe this is fine?)

In what other contexts do you think `MetricReader` can be used other than exporting? My understanding is it's always tied to exporting functionality in that it collects metrics from storage and exports.

Reply:
Any pull exporter e.g. PrometheusMetricReader. The spec requirement is specific to PushMetricExporter.

Reply:
Im ok with just adding this in in `periodicexportingmetricreader` for now. If the need comes in the future where there are more exporting metric readers, we can revisit. But not a huge opinion, I can go either way.

Reply:
I misunderstood the requirement. I interpreted it as that there should only be one ongoing export for any kind of implementation not just specific to PushMetricExporter. I am fine with this.